sprint.rb

```diff
@@ -20,27 +21,22 @@ def self.recent_sprints(count)
     sprints(as_of: nil).slice_after { |s| s.date >= Date.today }.first.last(count)
   end
 
-  def self.sprints(as_of: Date.today)
-    require "active_support/core_ext/numeric/time"
-    # The first date when we started doing this cadence
-    number = 76
-    date   = Date.parse("Jan 1, 2018")
-    range  = Date.parse("Dec 12, 2018")..date
+  def self.last_completed
+    Sprint.recent_sprints(2).first
+  end
 
-    as_of ||= date
+  def self.create_by_sprint_number(sprint_number)
+    SprintBoundaryIterator.new.each do |number, range|
+      return new(number, range) if number == sprint_number
+    end
+    nil
+  end
 
-    Enumerator.new do |y|
-      loop do
-        y << new(number, range) if date >= as_of
+  def self.sprints(as_of: Date.today)
+    as_of ||= SprintBoundaryIterator.start_range.end
 
-        last_date = date
-        number += 1
-        date += 2.weeks
-        while (date.month == 12 && (22..31).cover?(date.day)) || (date.month == 1 && (1..4).cover?(date.day))
-          date += 1.weeks
-        end
-        range = (last_date + 1.day)..date
-      end
+    SprintBoundaryIterator.new.collect do |number, range|
+      new(number, range) if range.end >= as_of
     end
```

Comment:
The SprintBoundaryIterator is a infinite iterator, so something needs to break here or it will look forever.

Reply:
The SprintBoundaryIterator#each method only emits [number, range] pairs in the past: `while today > range.first do`